### PR TITLE
feat: Added os.type pollyfill for Node std library

### PR DESCRIPTION
**Pull request by MVEMCJSUNPE**
Should check off os.type in #3802 . I know that there's already been work done by @jbergstroem on os.type in this [pull request](https://github.com/jbergstroem/deno/commit/b7b690ba6149dc701b5cb843bbec5b03c49e6c57) first, but for some reason a pull request wasn't submitted. If @jbergstroem wants to come forward and have his solution used, then I would be happy to oblige.

### CODE REVIEW

**Comment by jbergstroem** [comment]
> . If @jbergstroem wants to come forward and have his solution used, then I would be happy to oblige.

I context switched to other stuff and didn't follow up properly. Appreciate someone taking up the torch and crossing the goal line!

**Review by bnoordhuis — APPROVED** [review]
LGTM but the CI seems to think there's still a lint error somewhere.

**Comment by MVEMCJSUNPE** [comment]
I'm kind of stuck on where the lint error could be. From the error output, it seems like 'git status' is complaining?
```
2020-12-02T23:34:28.1997398Z dprint
2020-12-02T23:34:30.0319239Z rustfmt 103 file(s)
2020-12-02T23:34:31.4525905Z run tools/format.js
2020-12-02T23:34:31.4526506Z  M cli/ops/os.rs
```
This output corresponds to the lines in 'tools/format.js':
https://github.com/denoland/deno/blob/71ef5a9cd33390f7f41b80cdb57862b7f7507f76/tools/format.js#L38-L62

From looking at the code, it seems that 'git status' shouldn't be outputing anything.

**Comment by bartlomieju** [comment]
@MVEMCJSUNPE make sure to have latest stable Rust toolchain installed (1.48.0)

**Comment by MVEMCJSUNPE** [comment]
I'm pretty sure that's the version I have:
```sh
> rustc --version
rustc 1.48.0 (7eac88abb 2020-11-16)
```

**Comment by bartlomieju** [comment]
@MVEMCJSUNPE what's the version of `rustfmt` you're using? It should be `rustfmt 1.4.24-stable (eb894d53 2020-11-05)`

**Comment by MVEMCJSUNPE** [comment]
My rustfmt version:
```sh
> rustfmt --version
rustfmt 1.4.24-stable (eb894d5 2020-11-05)
```

**Comment by bartlomieju** [comment]
@MVEMCJSUNPE I'm not sure what's the problem, I'll reformat the code for you

**Comment by MVEMCJSUNPE** [comment]
Thanks @bartlomieju

**Comment by MVEMCJSUNPE** [comment]
For some reason, the license/cla is still pending even though I thought I signed it?

**Review by ry — COMMENTED** [review]
I don't think it's necessary to add a new op for this:
```
> deno eval -p Deno.build.os
linux
```

**Comment by MVEMCJSUNPE** [comment]
Now that @ry has mentioned Deno.build.os, I kind of agree. The only situation where I think that Deno.build.os might not be sufficient is if someone is [cross compiling](https://en.wikipedia.org/wiki/Cross_compiler) Deno. I'm not knowledgeable about how Deno.build.os is set, so I could be wrong, however.

**Comment by bartlomieju** [comment]
@MVEMCJSUNPE what do you mean by cross compiling Deno? I don't think there's an option to do that currently. Also `Deno.build` uses LLVM target triples for its values.

**Comment by MVEMCJSUNPE** [comment]
Oh nvm then. So should I use Deno.build.os to implement os.type polyfill?

**Comment by bartlomieju** [comment]
> Oh nvm then. So should I use Deno.build.os to implement os.type polyfill?

@MVEMCJSUNPE yes, that would be preferable solution

**Comment by MVEMCJSUNPE** [comment]
For some reason, the test for Mac OS X is failing.

**Comment by bartlomieju** [comment]
> For some reason, the check for Mac OS X is failing.

Looks like one-off failure.

File: cli/rt/30_os.js [hunk 1]
```diff
@@ -15,6 +15,10 @@
     return core.jsonOpSync("op_os_release");
   }
 
+  function osType() {
+    return core.jsonOpSync("op_os_type");
+  }
```
Comment (bartlomieju) [inline cli/rt/30_os.js:20]: Remove

File: cli/rt/30_os.js [hunk 2]
```diff
@@ -58,6 +62,7 @@
     execPath,
     exit,
     osRelease,
+    osType,
```
Comment (bartlomieju) [inline cli/rt/30_os.js:65]: Remove

**Review by bartlomieju — CHANGES_REQUESTED** [review]

**Review by ry — APPROVED** [review]
LGTM - thanks @MVEMCJSUNPE

**Pull request merged**